### PR TITLE
Updated ModelCheckpoint callback documentation

### LINKED ISSUE #6117: ModelCheckpoint is not saving top k models
## 🐛 Bug

`ModelCheckpoint` is not correctly monitoring metric values.

### To Reproduce

https://colab.research.google.com/drive/1onBmED7dngP_VwFxcFBMsnQi82KbizSk?usp=sharing

### Expected behavior

ModelCheckpoint should save top k models based on `x` metric, but it currently displays `Epoch XXX, step XXX: x was not in top 2` for every epoch.

### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.19.5
	- pyTorch_debug:     True
	- pyTorch_version:   1.7.0+cu101
	- pytorch-lightning: 1.2.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### Additional context

The documentation doesn't mention how one should set the metric to be used in `ModelCheckpoint`. Tried to use both `x` or `loss` value, but `ModelCheckpoint` shows the same message for both cases. Also, the message should be more clear, saying that `ModelCheckpoint` couldn't find chosen value to monitor instead of saying that it was not in top k, since it displays the same message if I choose to monitor some value that doesn't exist.

### PR DESCRIPTION
**Pull request by george-gca**
Signed-off-by: gca <george.gcac@gmail.com>

## What does this PR do?

Fixes #6117. It tries to clarify the documentation regarding monitoring metric values.

## Before submitting
- [X] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [X] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
 - [X] Is this pull request ready for review? (if not, please submit in draft mode)
 - [X] Check that all items from **Before submitting** are resolved
 - [X] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Always :)

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -38,7 +38,10 @@
 
 class ModelCheckpoint(Callback):
     r"""
-    Save the model after every epoch by monitoring a quantity.
+    Save the model after every epoch by monitoring a quantity. Every metric logged with
+    :meth:`~pytorch_lightning.core.lightning.log` or :meth:`~pytorch_lightning.core.lightning.log_dict` in
+    LightningModule is a candidate for the monitor key. For more information, see
+    :ref:`common/weights_loading:Checkpoint saving`.
```
Comment (awaelchli) [inline pytorch_lightning/callbacks/model_checkpoint.py:44]: not 100% sure, but I think we can only reference sections by title within the same document. 
You probably have to refer to just :ref:`_weights_loading`

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 2]
```diff
@@ -38,7 +38,10 @@
 
 class ModelCheckpoint(Callback):
     r"""
-    Save the model after every epoch by monitoring a quantity.
+    Save the model after every epoch by monitoring a quantity. Every metric logged with
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/model_checkpoint.py:41]: Is this an old version of Lightning? the "after every epoch" isn't true - there are more options being added for greater flexibility. See https://github.com/PyTorchLightning/pytorch-lightning/blob/19e67d18c472c3a03dec4dd9bfcef031e9ca8719/pytorch_lightning/callbacks/model_checkpoint.py#L96-L107

Comment (george-gca) [reply pytorch_lightning/callbacks/model_checkpoint.py:44]: Actually it was like that in the documentation, that's why I didn't change the `after every epoch` part. But you are right. What do you think about this before the part that I added:

Save the model after every ``every_n_train_steps`` training steps or every ``every_n_val_epochs`` validation epochs by monitoring a quantity. These options are mutually exclusive.

Comment (awaelchli) [reply pytorch_lightning/callbacks/model_checkpoint.py:44]: I would not mention the arguments before they are introduced later on. 
How about:

Save the model periodically by monitoring a quantity. ...

**Review by ananthsub — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
